supplierProduct/models.py

```diff
@@ -0,0 +1,65 @@
+from django.db import models
+from django.core.validators import MinLengthValidator, MinValueValidator
+from product.models import Product
+
+
+class User(models.Model):
+    user_name = models.CharField(max_length=32,
+                                 primary_key=True,
+                                 null=False,
+                                 validators=[MinLengthValidator(6)],
+                                 help_text="User Name")
+    first_name = models.CharField(max_length=32,
+                                  validators=[MinLengthValidator(1)],
+                                  help_text="First Name")
+    last_name = models.CharField(max_length=32,
+                                 validators=[MinLengthValidator(1)],
+                                 help_text="Last Name")
+    password = models.CharField(max_length=16,
+                                validators=[MinLengthValidator(6)],
+                                null=False,
+                                help_text="Password")
+    user_type = models.CharField(max_length=16,
+                                 validators=[MinLengthValidator(6)],
+                                 null=False,
+                                 choices=[('client', 'client'), ('supplier', 'supplier')],
+                                 help_text="User Type - supplier/client")
+
+    class Meta:
+        abstract = True
+
+
+class Supplier(User):
+    """Supplier class inherit from User abstract class.
+    Supplier class lets the app user to do variety of actions
+    which evolves the Product and Delivery class.
+    Attributes:
+        business_name: a string which holds the Supplier business name.
+    """
+    business_name = models.CharField(max_length=32,
+                                     validators=[MinLengthValidator(6)],
+                                     null=False,
+                                     help_text="Business Name")
+
+
+class supplierProduct(models.Model):
+    suppler_product_id = models.IntegerField(primary_key=True)
```

Comment:
Delete this field because django has its default autofield primary key and we dont want to override it

when you delete it update the migrations and the functions and tests if needed

Reply:
i didn't quite get your comment , 
you're think  I'm suppose  to delete the primary "key = True" ?

Reply:
You don't need to delete it. 
According to the ERD of your app, you decided to give a special name for the primary key of this table (therefore you have to declare it).

Reply:
In the erd we were more informative, i know things needs to be identical but if you delete the supplier id that you are using and use the default id it will be more understandable and more comfortable to use in the future

Reply:
If that's the case then I'm ok with that change. 
But we need to make sure everyone agrees and aware of that change, and a good way to do it is to create a PR updating the ERD.